### PR TITLE
Add sed command to vim

### PR DESCRIPTION
**Pull request by pleshevskiy**

### CODE REVIEW

File: plugins/c9.ide.ace.keymaps/cli.js [hunk 1]
```diff
@@ -187,6 +187,122 @@ define(function(require, exports, module) {
...
+        }
+        
+        cliCmds[':'].reCommands = {
+            /**
+             * @see {@link http://vim.wikia.com/wiki/Search_and_replace|Vim wiki - sed}
+             */
+            'sed': {
+                regex: /^(%|'<,'>|(?:\d+|\.),(?:\+?\d+|\$|\.))?s(\/|#)(.*?)\2(.*?)\2([giIc]*)$/,
+                action: function (editor, cmd, data) {
+                    var pattern = (
```
Comment (nightwing) [inline plugins/c9.ide.ace.keymaps/cli.js]: have you seen the implementation in vim mode, if yes how does this compare with it?  https://github.com/ajaxorg/ace/blob/master/lib/ace/keyboard/vim.js#L5266

File: plugins/c9.ide.ace.keymaps/cli.js [hunk 2]
```diff
@@ -187,6 +187,122 @@ define(function(require, exports, module) {
...
+                            
+                            return replacement.replace(/\\(\d+)/g, function (m, m1) {
+                                var i = parseInt(m1, 10);
+                                return i < args.length - 2 ? args[i] : '';
+                            });
+                        });
+                    }
+                    
+                    if (lastReplaceRange) {
+                        editor.setValue(parts.join('\n'), -1);
```
Comment (nightwing) [inline plugins/c9.ide.ace.keymaps/cli.js]: replacing this way will cause problems with undo history and selection. I think we need to either use the api from https://github.com/ajaxorg/ace/blob/master/lib/ace/search.js or https://github.com/c9/core/blob/master/plugins/c9.ide.find.replace/findreplace.js#L983

File: plugins/c9.ide.ace.keymaps/cli.js [hunk 3]
```diff
@@ -357,26 +473,37 @@ define(function(require, exports, module) {
...
+            else if (commands.commands[firstCmd]) {
+                commands.exec(firstCmd, null, { argv: args });
             }
             else if (/^\d+$/.test(cmd)) {
                 ed.gotoLine(cmd, 0, true);
             }
             else {
-                ed.cmdLine.setTimedMessage("Vim command '" 
-                    + cmd + "' not implemented.", 1500);
+                for (var key in this.reCommands) {
```
Comment (nightwing) [inline plugins/c9.ide.ace.keymaps/cli.js]: are there any other reCommands that we'll need to add here in the future

File: plugins/c9.ide.ace.keymaps/cli.js [hunk 4]
```diff
@@ -357,26 +473,37 @@ define(function(require, exports, module) {
...
+                    var reCmd = this.reCommands[key];
+                    var match = reCmd.regex.exec(cmd);
+                    if (match) {
+                        return reCmd.action(ed, cmd, {
+                            'match': match,
+                            'argv': cmd.split(match[0], 1).slice(-1)[0].split(/\s+/)
```
Comment (nightwing) [inline plugins/c9.ide.ace.keymaps/cli.js]: it's better to not add quotes to object keys when possible.

**Comment by pleshevskiy** [comment]
I send pull request to vim https://github.com/ajaxorg/ace/pull/3484

File: plugins/c9.ide.ace.keymaps/cli.js (hunk 3 above)
Comment (pleshevskiy) [reply plugins/c9.ide.ace.keymaps/cli.js]: All commands with range (Delete, Yank, Read/Save) and i hope add shell commands

**Comment by nightwing** [comment]
Merged in https://github.com/c9/core/commit/eb710ca024a492043dab19a6535dc2348eef7bf4, thanks!

**Pull request closed without merging**